rust/worker/src/segment/test.rs

```diff
@@ -1,20 +1,26 @@
 use std::sync::atomic::AtomicU32;
 
 use chroma_blockstore::{provider::BlockfileProvider, test_arrow_blockfile_provider};
-use chroma_types::{test_segment, Chunk, LogRecord, OperationRecord, Segment, SegmentScope};
-use indicatif::ProgressIterator;
+use chroma_index::{hnsw_provider::HnswIndexProvider, test_hnsw_index_provider};
+use chroma_types::{
+    test_segment, Chunk, Collection, CollectionUuid, LogRecord, OperationRecord, Segment,
+    SegmentScope,
+};
 
-use crate::log::test::LogGenerator;
+use crate::log::test::{LogGenerator, TEST_EMBEDDING_DIMENSION};
 
 use super::{
     metadata_segment::MetadataSegmentWriter, record_segment::RecordSegmentWriter, LogMaterializer,
     SegmentFlusher, SegmentWriter,
 };
 
 pub struct TestSegment {
-    pub blockfile_provider: BlockfileProvider,
+    pub hnsw: HnswIndexProvider,
+    pub blockfile: BlockfileProvider,
```

Comment:
i prefer blockfile_provider and hnsw_provider

Reply:
Renamed